operator/internal/central/extensions/reconcile_pvc.go

```diff
@@ -200,7 +287,31 @@ func (r *reconcilePVCExtensionRun) handleDelete() error {
 }
 
 func (r *reconcilePVCExtensionRun) handleCreate(claimName string, pvcConfig *platform.PersistentVolumeClaim) error {
-	size, err := parseResourceQuantityOr(pvcConfig.Size, defaultPVCSize)
+	// Before creating a PVC, verify if prerequisites are met. Currently there
+	// is only one requirement, a default storage class must exists or a
+	// storage class has to be specified explicitly. Since it's highly specific
+	// for PVCs only, it's implemented inside the extension, instead of
+	// collecting this information at the start and passing it into the
+	// extension.
+	//
+	// Note that to make this check less disruptive, in case if we face an
+	// error we still try to create a PVC.
+	hasDefault, err := utils.HasDefaultStorageClass(r.ctx, r.client)
+	if err != nil {
+		r.log.Error(err, fmt.Sprintf("cannot find the default storage class, but proceeding with %q PVC creation", claimName))
```

Comment:
I'm a little uneasy about this... If there is no default storage class, but we face an intermittent network/kube API availability issue, then we're going to create the backup PVC anyway.

Reply:
Well, I don't mind turn it other way around and do not proceed in that scenario.

Reply:
I guess the result of `HasDefaultStorageClass` has a little troubleshooting value, so it makes sense to call it unconditionally. But then:
1. For the main volume, just log the result and proceed anyway
2. For the backup volume, return error on `err != nil`, return nil on `!hasDefault`, and proceed otherwise
WDYT?

Reply:
Makes sense.